### PR TITLE
A handful of cleanups in keepassxc-cli

### PR DESCRIPTION
**Pull request by sjamesr**
## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ Minor cleanup

## Description and Context
* empty destructors are replaced with default destructors
* a few loop variables made into const references to avoid copies
* add missing `override` spec for some `Command::execute` methods

## Testing strategy
Ran unit tests

## Checklist:
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/cli/Create.h [hunk 1]
```diff
@@ -27,12 +27,11 @@ class Create : public Command
...
 public:
     Create();
-    ~Create();
-    int execute(const QStringList& arguments);
+    ~Create() override = default;
+    int execute(const QStringList& arguments) override;
 
 private:
     QSharedPointer<PasswordKey> getPasswordFromStdin();
-    QSharedPointer<FileKey> getFileKeyFromStdin();
```
Comment (louib) [inline src/cli/Create.h]: @sjamesr good catch!

File: src/cli/Edit.h [hunk 2]
```diff
@@ -24,8 +24,8 @@ class Edit : public DatabaseCommand
 {
 public:
     Edit();
-    ~Edit();
-    int executeWithDatabase(QSharedPointer<Database> db, QSharedPointer<QCommandLineParser> parser);
+    ~Edit() override = default;
```
Comment (louib) [inline src/cli/Edit.h]: @sjamesr is there a reason why you decided to remove the destructor declaration in some commands, and declared it `override = default` in others?

Comment (sjamesr) [reply src/cli/Edit.h]: Oh, that was just an oversight, should be fixed now. Thanks!

**Comment by louib** [comment]
@droidmonkey this one is ready to merge I believe!

File: src/cli/Add.cpp [hunk 3]
```diff
@@ -62,10 +62,6 @@ Add::Add()
     positionalArguments.append({QString("entry"), QObject::tr("Path of the entry to add."), QString("")});
 }
 
-Add::~Add()
-{
-}
```
Comment (phoerious) [inline src/cli/Add.cpp]: I believe empty constructors were necessary in certain situations where Qt smart pointers were used. But if it compiles and CI doesn't complain, I'm okay with removing them.

File: src/cli/Command.cpp [hunk 4]
```diff
@@ -80,14 +80,14 @@ QSharedPointer<QCommandLineParser> Command::getCommandLineParser(const QStringLi
 
     QSharedPointer<QCommandLineParser> parser = QSharedPointer<QCommandLineParser>(new QCommandLineParser());
     parser->setApplicationDescription(description);
-    for (CommandLineArgument positionalArgument : positionalArguments) {
+    for (const CommandLineArgument& positionalArgument : positionalArguments) {
```
Comment (phoerious) [inline src/cli/Command.cpp]: QString does implicit sharing, but using references is consistent with our coding style.

**Pull request merged**